test/basic_functionality.jl

```diff
@@ -5,12 +5,13 @@ m,n,k,s = 1000,1000,5,1000*1000;
 # matrix to encode
 X_real, Y_real = randn(m,k), randn(k,n);
 A = X_real*Y_real;
-losses = fill(quadratic(),n);
+losses = convert(Array{Loss,1}, fill(quadratic(),n));
```

Comment:
it's really gross that we need to perform this conversion explicitly. but i can't think of a good way to get around it...